File: Leetcode/python/2530-max-score-after-k-operations.py
from typing import List
import heapq
import math

class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
            res = 0
            maxHeap = [-n for n in nums]
            heapq.heapify(maxHeap)

            while k:
                n = -heapq.heappop(maxHeap)
                res += n
                k -= 1
                heapq.heappush(maxHeap, -math.ceil(n / 3))

            return res

File: Leetcode/typescript/2530-max-score-after-k-operations.ts
function maxKelements(nums: number[], k: number): number {
  let res = 0;

  const maxHeap = new MaxPriorityQueue({ priority: (x) => x });

  for (const n of nums) {
    maxHeap.enqueue(n);
  }
  while (k > 0) {
    const n = maxHeap.dequeue().element;
    res += n;
    k--;
    maxHeap.enqueue(Math.ceil(n / 3));
  }

  return res;
}
